porch/repository/pkg/cache/repository.go

```diff
@@ -138,7 +165,9 @@ func (r *cachedRepository) CreatePackageRevision(ctx context.Context, obj *v1alp
 }
 
 func (r *cachedRepository) UpdatePackage(ctx context.Context, old repository.PackageRevision) (repository.PackageDraft, error) {
-	created, err := r.repo.UpdatePackage(ctx, old)
+	// Unwrap
+	unwrapped := old.(*cachedPackageRevision).PackageRevision
```

Comment:
Why do we need to unwrap here? It seems like the `UpdatePackage` accepts a parameter of type `PackageRevision` which the `*cachedPackageRevision` implements.

Reply:
the Git repository will cast the interface to *gitPackageRevision.

We could expand the interface and avoid the cast but Git is getting at some fields not exposed via the interface.